TU20Bot/Configuration/Controllers/AuthenticationController.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using EmbedIO;
+using EmbedIO.Routing;
+using EmbedIO.WebApi;
+using Jose;
+using MongoDB.Driver;
+using TU20Bot.Models;
+
+namespace TU20Bot.Configuration.Controllers {
+    public class AuthenticationController : ServerController {
+
+        [Route(HttpVerbs.Post, "/login")]
+        public string Login([QueryField] string username, [QueryField] string password) {
+
+            var user = server.client.database.GetCollection<AccountModel>(AccountModel.collectionName).Find(_user => _user.username.Equals(username.ToLower()));
+
+            if (!user.Any()) {
+                throw HttpException.Unauthorized();
+            }
+
+            if (user.First().checkPassword(password)) {
+                return JWT.Encode(new AuthorizationPayload {
+                    fullName = username,
+                    validUntil = DateTime.Now.AddHours(1), // Expire in 1 hour
+                    permissions = user.First().permissions
+                }, Encoding.UTF8.GetBytes(server.config.jwtSecret), JwsAlgorithm.HS256);
+            }
+
+            throw HttpException.Unauthorized();
```

Comment:
This isn't a big deal, but I like to think that the default path of the function will always end in success.
```suggestion
            if (!user.First().checkPassword(password))
                throw HttpException.Unauthorized();
            
            return JWT.Encode(new AuthorizationPayload {
                fullName = username,
                validUntil = DateTime.Now.AddHours(1), // Expire in 1 hour
                permissions = user.First().permissions
            }, Encoding.UTF8.GetBytes(server.config.jwtSecret), JwsAlgorithm.HS256);
```